### DISCUSSION TITLE
Finetune pretrained DPA-1 model with dpgen

### DISCUSSION THREAD

**Discussion by KeweiZhao** [deepmodeling/dpgen#1116] [Q&A] [answered]
Hello,

I'm wondering if there is a way to finetune a pretrained DPA-1 model with dpgen (e.g. OC_10M.pb). I haven't find a way to pass the "--finetune" argument to dp using dpgen.

Thanks,
Kewei

**Comment by wanghan-iapcm** [comment, marked as answer]
please refer to this discussion https://github.com/deepmodeling/deepmd-kit/discussions/2249

**Reply by KeweiZhao** [reply]
Thanks for the answer. I could train a model by finetuning the pretrained OC_10M.pb model using the deepmd-kit 2.2.0beta without any problem. I'm wondering **if the "finetune" function can be called from dpgen** so that the training process can be iterative (i.e. how to apply the "--finetune OC_10M.pb" argument in dpgen). Or should I first accumulate data with dpgen then use the finetune function for the last training before generating a model for production run?

**Reply by wanghan-iapcm** [reply]
You may provide the OC_10M.pb as [user_forward_files](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/mdata.html#argument:run_mdata/train/user_forward_files),  and pass the finetune option by [train/command](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/mdata.html#argument:run_mdata/train/command)

**Reply by KeweiZhao** [reply]
Thanks. This should work but I haven't tested, as the "--finetune" options comes after "dp train param.json", therefore is not very straightforward to put it into the "train/command" str.

An alternative way (tested) is to first use dp to finetune a pretrained model manually, put the .pb file into iter.000000/00.train folder and modify the record.dpgen file (up to "0 2"), then run dpgen. This way dpgen will skip the training in the first iteration and use the model provided to run model_devi.

**Reply by wanghan-iapcm** [reply]
Good point, it is not supported yet. I change discussion to a feature request issue.

**Reply by niuniu800** [reply]
Can the DPA-1 pre-trained model be fine-tuned with new data, and is it now possible to combine it with DPGEN for use?
